docs/docs/09-rfcs/2023-03-08-wing-preview-environment.md

```diff
@@ -0,0 +1,329 @@
+6. If a Wing Cloud singed-in user that doesn't have access to the preview environment's repository attempts to access a preview environment, they will be directed to the Wing Cloud dashboard page and will be notified that they don't have access to the preview environment's repository.
+
+#### Wing Cloud Dashboard
+
+1. Once signed in, you will be redirected to the Wing Cloud dashboard page.
+2. From Wing Cloud dashboard you can install Wing Cloud application on any of your repositories.
+3. Wing Cloud dashboard page display a list of all the repositories that you have installed Wing Cloud application on.
+4. For each repository, there is a list of all the preview environments.
+5. In order to see a preview environment, click on the preview environment url.
+
+#### Installation of Wing Cloud application
+
+In order to use Wing Cloud Preview Environments, you should sign in to Wing Cloud and install Wing Cloud application in your GitHub account and repository.
+The installation process is straightforward using GitHub application:
+
+1. Goto https://wing.cloud and sign in
+2. If you are already signed in, you will be redirected to the Wing Cloud dashboard page https://wing.cloud/dashboard
+3. In https://wing.cloud/dashboard, Click on "Deploy with Wing Cloud"
+4. Complete GitHub application authentication.
+5. Grant the repositories you want to have preview environments for, access permission.
+6. You are now ready to use Wing Cloud Preview Environments.
+7. A welcome email will be sent to you with additional information and links to documentation.
+
+#### Preview Environment for production branch
+
+For the repository production branch, there is a single preview environment that will constantly be up-to-date with the latest code committed.
+A link to the preview environment is available in the repository main page ("About" section).
+Production branch environment url structure is: `https://<gh-repository>-<gh-branch>-<entry-point>.wingcloud.app`
+Once the installation of the Wing Cloud application is completed, an initial preview environment will be created and will be available in the repository main page.
+
+#### Preview Environments for Pull Requests
+
+Upon each creation of a pull request an automatic comment will be added to the PR and will guide the developer to dedicated preview environments.
+provide a streamlined process for updating a pull request preview environment, each PR code changes will redeploy the preview environment and will also:
+
+1. Provide a real-time build and deployment status updates
+2. While preview environment is being updated, the user that is currently using the environment will be notified and will be able to refresh the page to get the latest changes.
+3. Each entry point will have a seperated preview environment
+4. Ensure a consistent and unique url for each preview environment. The url structure: `https://<gh-repository>-<gh-branch>-<entry-point>.wingcloud.app`
```

Comment:
Will we have the option to also have a URL per commit like Vercel does?

Will the 'entry-point' be the full path of the file? (will not be unique otherwise)

I think <entry-point>-<gh-branch>-<repo>.wingcloud.app will be more readable. 

For example:
```
entry1-add-new-feat-my-app.wingcloud.app
entry2-add-new-feat-my-app.wingcloud.app
```

vs

```
my-app-add-new-feat-entry1.wingcloud.app
my-app-add-new-feat-entry1.wingcloud.app
```